File: BackEnd/Backend/Backend/Models/ForumComment.cs
using System;
namespace Backend.Models
{
    public class ForumComment
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public ForumPost? ForumPost { get; set; }
        public int ForumPostId { get; set; } 
        public User? User { get; set; }
        public DateTime Date { get; set; }
        public string CommentText { get; set; } = string.Empty;
    }
}

File: frontend/LicitatieFrontend/src/app/Models/forum-comment/forum-comment.ts
export class ForumCommentDto {
    id!: number;
    userId!: number;
    userName!: string;
    forumPostId!: number; 
    date!: string;          
    commentText!: string; 
    
    constructor(init?: Partial<ForumCommentDto>) {
        Object.assign(this, init);
    }
}

export interface CreateForumCommentDto {
    forumPostId: number;
    userId: number;
    commentText: string;
}

export interface UpdateForumCommentDto {
    commentText: string;
}

// Pastram clasa veche ca un "alias" pentru a nu strica imediat paginile la care lucreaza colegul
export class ForumComment extends ForumCommentDto {}